moexmen/superlogger
# Add missing lock file and Ruby version for reproducible builds

## Context

Both the lock file and the Ruby version are not specified in the repository. As a result, the dependencies that are installed when running `bundle update` will always be the latest versions which do not work with the testing setup. This is because as Superlogger's dependencies are updated, so do their dependencies and other configuration which are incompatible with the current testing setup.

## Changes

In light of upcoming functionality changes, we want to be able to run the Superlogger test suite. This PR aims to achieve that by adding a lock file and Ruby version that have been verified to work, thus allowing for reproducible builds (and thus a working test command).

## How to Verify

1. Check out the branch associated with this PR.
1. Install the version of Ruby specified in `.ruby-version`.
1. Install the dependencies specified in the lock file via `bundle update`.
1. Run `rake test`.

The test suite should be able to run, albeit with failures. These test failures will be resolved in a later PR.
